Measurements/ReferenceImplementation.qs

```diff
@@ -371,7 +383,7 @@ namespace Quantum.Kata.Measurements {
 
         // Apply diag(..) (H ⊗ H) diag(..)
         within {
-            ApplyDiag(qs);
+            ApplyDiag_Reference(qs);
```

Comment:
Theoretically this could also be implemented using a library operation, but I'm not sure using it will be more readable (and more instructive!) for the learner. What do you think?

```suggestion
            ApplyDiagonalUnitary([PI(), 0.0, 0.0, 0.0], LittleEndian(qs));
```

Reply:
i like this suggestion.  but if i hit "commit suggestion" i am going to leave the ApplyDiag_Reference orphaned in the code, right?  so i should change locally and push changes up, correct?

Reply:
Yes, you'll need to do corresponding changes locally, suggestions don't play nicely when the same suggestion applies in multiple places or needs matching changes elsewhere.